_specifications/lsp/3.18/language/documentSymbol.md

```diff
@@ -165,11 +165,103 @@ export namespace SymbolTag {
 
 	/**
 	 * Render a symbol as obsolete, usually using a strike-out.
+	 * @since 3.16
 	 */
-	export const Deprecated: 1 = 1;
+	export const Deprecated = 1;
+
+    /**
+	 * Render a symbol with visibility / access modifier "private".
+	 * @since 3.18
+	 */
+    export const Private = 2;
+    
+    /**
+	 * Render a symbol with visibility "package private", e.g. in Java.
+	 * @since 3.18
+	 */
+    export const Package = 3;
+    
+    /**
+	 * Render a symbol with visibility / access modifier "protected".
+	 * The modifier could be combined e.g. with "internal" or "private" in languages like C#.
+	 * @since 3.18
+	 */
+    export const Protected = 4;
+    
+    /**
+	 * Render a symbol with visibility / access modifier "public".
+	 * @since 3.18
+	 */
+    export const Public = 5;
+    
+    /**
+	 * Render a symbol with visibility / access modifier "internal", e.g. in C# or Kotlin.
+	 * @since 3.18
+	 */
+    export const Internal= 6;
+    
+    /**
+	 * Render a symbol with visibility / access modifier "file", e.g. in C#.
+	 * @since 3.18
+	 */
+    export const File = 7;
+
+	/**
+	 * Render a symbol as "static".
+	 * @since 3.18
+	 */
+    export const Static = 8;
+    
+    /**
+	 * Render a symbol as "abstract".
+	 * @since 3.18
+	 */
+    export const Abstract = 9;
+    
+    /**
+	 * Render a symbol as "final".
+	 * @since 3.18
+	 */
+    export const Final = 10;
+
+    /**
+	 * Render a symbol as "sealed", e.g. classes and interfaces in Kotlin.
+	 * @since 3.18
+	 */
+    export const Sealed = 11;
+    
+    /**
+	 * Render a symbol as "transient", e.g. in Java.
+	 * @since 3.18
+	 */
+    export const Transient = 12;
+    
+    /**
+	 * Render a symbol as "volatile", e.g. in Java.
+	 * @since 3.18
+	 */
+    export const Volatile = 13;
+    
+    /**
+	 * Render a symbol as "synchronized", e.g. in Java.
+	 * @since 3.18
+	 */
+    export const Synchronized = 14;
+    
+    /**
+	 * Render a symbol as "virtual", e.g. in C++.
+	 * @since 3.18
+	 */
+    export const Virtual = 15;
```

Comment:
Const for C++ would be useful, too

Reply:
I [reconsidered](https://github.com/llvm/llvm-project/pull/113669#issuecomment-2444643427) a _Const_ tag and I think we can use the _ReadOnly_ tag for that purpose. Having both tags, _Const_ and _ReadOnly_ might be confusing since they mean the same in principle. Besides, I don't think, we should use many language-specific keywords as tags in LSP specification. Instead, we should describe the concepts behind them.